src/software/new_geom/polynomial1d.h

```diff
@@ -0,0 +1,153 @@
+#pragma once
+
+#include <cmath>
+#include <vector>
+
+/**
+ * Polynomial1d is a representation of a polynomial
+ * that can calculate values at a given input
+ */
+class Polynomial1d
+{
+   public:
+    /**
+     * Construct a zero polynomial
+     */
+    explicit Polynomial1d();
+
+    /**
+     * Construct a polynomial from coefficients of the form
+     * coeffs[0] + coeffs[1]*x^(1) + ... + coeffs[n-1]*x^(n-1)
+     *
+     * @param coeffs coefficients of the polynomial
+     */
+    explicit Polynomial1d(const std::vector<double> &coeffs);
+
+    /**
+     * Construct a polynomial from coefficients of the form
+     * coeffs[0] + coeffs[1]*x^(1) + ... + coeffs[n-1]*x^(n-1)
+     *
+     * @param coeffs coefficients of the polynomial
+     */
+    explicit Polynomial1d(const std::initializer_list<double> &coeffs);
+
+    /**
+     * Construct a linear polynomial from two pairs of input/output
+     *
+     * @param constraint1 first pair of values: first is the input, second is the output
+     * @param constraint2 second pair of values: first is the input, second is the output
+     *
+     * @throws std::invalid_argument if constraint1.first == constraint2.first
+     */
+    static Polynomial1d constructLinearPolynomialFromConstraints(
+        const std::pair<double, double> &constraint1,
```

Comment:
Any reason not to use `Point` here? I'm guessing you may have done this to keep it more mathematical and keep away of spatial representations?

If we do keep it as points, may be convenient to have an easy to way to convert `Point` to `pair<double, double>` since I mostly see this getting used with points (not necessarily something for this PR).

Reply:
A constraint is a pair of (x,f(x)). For clarity, can we either use a new datatype to represent this or break it apart, i.e. `constructLinearPolynomialFromConstraints(double t1, double x_at_t1, double t2, double x_at_t2)`

Reply:
Yeah, since it's a 1d polynomial a "Point" doesn't really make much sense. 

Changed to `double input_1, double output_1, double input_2, double output_2` to be super explict.